hangman.py

```diff
@@ -0,0 +1,129 @@
+    """
+    user_guess = input("Please guess a letter: ").upper()
+    hangman_clear_screen()
+    return user_guess
+
+
+def hangman_clear_screen():
+    """
+    Clears screen based on the range specifed.
+    """
+    for i in range(1):
+        print("\n")
+
+
+def hangman_win_lose(word, guessed):
+    """
+    Checks if the player won or lost and prints the information
+    """
+    if guessed:
+        print("Congratulations! You won the game!")
+    else:
+        print("Sorry you ran out of lives! The correct answer is {0} ".format(word))
+    return 
+
+
+def hangman_algorithm(theme, word):
+    """
+    Sets user attemps and initial state of the game. Obtains user guesses and
+    checks for validity. If user guess is correct, displays the appropriate
+    letters corresponding to the guess. If the user guess is invalid, decreases
+    attempts and informs user. If user guessed a previously guessed letter, no
+    penealty for the user.
+    """
+    user_guess_list = set()
+    attempts = 9
+    guessed = False
+    valid_indices = []
+
+    print_dashes = hangman_print_dashes(word)
+    hangman_generate_hint(theme, word)
+
+    print_dashes_list = list(print_dashes)
+    word_list = list(word)
+
+    while not guessed and attempts > 0:
+        print("Remaining Lives: {0} ".format(attempts))
+        print("Letters Guessed: {0} ".format(user_guess_list))
+        user_guess = hangman_get_letter()
+
+        if (len(user_guess) == 1 and user_guess.isalpha()):
+            if user_guess in user_guess_list:
+                print("You've already guessed this letter! Try again!")
+                print("Word: {0}".format(" ".join(print_dashes)))
+                hangman_generate_hint(theme, word)
+            elif user_guess not in word:
+                print("Sorry! {0} is not in the word. Try again!".format(user_guess))
+                user_guess_list.add(user_guess)
+                attempts -= 1
+                print("Word: {0}".format(" ".join(print_dashes)))
+                hangman_generate_hint(theme, word)
+            else:
+                print("This letter is in the word!")
+                user_guess_list.add(user_guess)
+
+                print_dashes = " ".join(letter if letter in user_guess_list
+                        else "_" for letter in word)
+
+                print("Word: {0}".format(print_dashes))
+                hangman_generate_hint(theme, word)
+
+                if "_" not in print_dashes:
+                    guessed = True
+
+        else:
+            print("Please enter a valid, single letter.")
+            print("Word: {0}".format(" ".join(print_dashes)))
+            hangman_generate_hint(theme, word)
+
+    hangman_win_lose(word, guessed)
+
+
+def main():
+    replay_trigger = "Y"
+
+    theme, word = hangman_pick_word()
+    hangman_algorithm(theme, word)
+
+    replay_question = input("Press Y to replay the game.\
+            Press any other key to exit game.")
```

Comment:
Pressing the Y key doesn't start a new game, but entering Y and pressing the Enter key does. Also, "any other key" could be perceived as a non-character key (F1, F2, Backspace etc) which won't actually end the game and would be a bug.

An example fix would be `"Enter Y to replay game. Enter N to exit."`

Reply:
Yeah thats true. I'll update with your wording as it is indeed clearer!